controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,253 @@
+package storagecluster
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/yaml"
+)
+
+type ConnectedClient struct {
+	NamespacedName types.NamespacedName `yaml:"metadata"`
+	ClusterID      string               `yaml:"clusterId"`
+}
+type InfoStorageCluster struct {
+	NamespacedName          types.NamespacedName `yaml:"metadata"`
+	StorageProviderEndpoint string               `yaml:"storageProviderEndpoint"`
+	CephClusterFSID         string               `yaml:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
+	OdfVersion          string             `yaml:"odfVersion"`
+	OdfDeploymentType   string             `yaml:"odfDeploymentType"`
+	Clients             []ConnectedClient  `yaml:"clients"`
+	StorageCluster      InfoStorageCluster `yaml:"storageCluster"`
+	StorageClusterCount int                `yaml:"storageClusterCount"`
+	StorageSystemName   string             `yaml:"storageSystemName"`
+	IsDROptimized       bool               `yaml:"isDROptimized"`
+}
+
+const (
+	OdfInfoKeyName            = "config.yaml"
```

Comment:
This string represent the key suffix not the complete name
```suggestion
	OdfInfoKeySuffix            = "config.yaml"
```